backfiller/backfiller/main.py

```diff
@@ -98,6 +124,14 @@ def get_remote_segment(base_dir, node, channel, quality, hour, missing_segment,
 		with open(temp_path, 'w') as f:
 			for chunk in resp.iter_content(8192):
 				f.write(chunk)
+				hash.update(chunk)
+
+		hash_str = b64encode(hash.digest(), "-_").rstrip("=")
+		if hash_str not in missing_segment:
```

Comment:
I'm deeply suspicious of this line, but I can't come up with a scenario where it breaks for now.
What we should really be doing is passing in the parsed `segment` object to this function, instead of all the piecemeal components. then we can just compare to `segment.hash`.

Reply:
Hopefully what I've just pushed does this in a more elegant and less error prone way.